api/config/v1alpha1/metric_types.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type ProxyMetric struct {
+	// Prometheus defines the configuration for Admin endpoint `/stats/prometheus`.
+	Prometheus *PrometheusProvider `json:"prometheus,omitempty"`
```

Comment:
should we create a `Push` / `Pull` based hierarchy here ?
```
telemetry:
  metrics:
    type: Pull
    pull:
      type: Prometheus
      prometheus: 
```

```
telemetry:
  metrics:
    type: Push
    push:
      type: OpenTelemetry
      openTelemetry: 
```

Reply:
Actually, they can exist at the same time.

Reply:
ah I see

Reply:
if thats the case do you think this format is handy or too wordy ?
this way we allow the user to specify both push and pull but only one for each type
```
telemetry:
  metrics:
    push:
      type: OpenTelemetry
      openTelemetry: 
    pull:
      type: Prometheus
      prometheus: 
```

Reply:
@AliceProxy @kflynn and I discussed this in the community meeting today and feel like `pull` and `push` might be redundant keywords to type, resolving this comment